src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProject.cs

```diff
@@ -273,14 +284,14 @@ private void ChangeProjectProperty<T>(ref T field, T newValue, Action<SolutionCh
 
             if (_activeBatchScopes > 0)
             {
-                _projectPropertyModificationsInBatch.Add(solutionChanges => updateSolution(solutionChanges, oldValue));
+                _projectPropertyModificationsInBatch.Add(
+                    (solutionChanges, projectUpdateState) => updateSolution(solutionChanges, projectUpdateState, oldValue));
             }
             else
             {
-                _projectSystemProjectFactory.ApplyBatchChangeToWorkspace(solutionChanges =>
-                {
-                    updateSolution(solutionChanges, oldValue);
-                });
+                _projectSystemProjectFactory.ApplyBatchChangeToWorkspace(
+                    (solutionChanges, projectUpdateState) => updateSolution(solutionChanges, projectUpdateState, oldValue),
+                    onAfterUpdate: onAfterUpdate);
```

Comment:
so, i see you passing in onafterupdate here.  but in the `true` part of the `if-statement` it seems to be dropped.  I'm not sure i understand what the implications are of that.

Reply:
yup, that would be an issue.  no one actually ever passed an after update function here though so I can remove this.